polynote-kernel/src/main/scala/polynote/kernel/util/KernelContext.scala

```diff
@@ -87,11 +87,18 @@ final case class KernelContext(global: Global, classPath: List[File], classLoade
   def reprsOf(value: Any, typ: global.Type): List[ValueRepr] = {
     val otherReprs = try {
       reprsOfCache.getOrElseUpdate(typ,
-        runtimeTools.inferImplicitValue(ru.appliedType(ru.typeOf[ReprsOf[_]].typeConstructor, importToRuntime.importType(typ))) match {
-          case ru.EmptyTree => ReprsOf.empty
-          case tree =>
-            val untyped = runtimeTools.untypecheck(tree)
-            runtimeTools.eval(untyped).asInstanceOf[ReprsOf[Any]]
+        {
+          val run = new global.Run()
+          val appliedType = global.appliedType(global.typeOf[ReprsOf[_]].typeConstructor, typ)
+
+          val runtimeType = importToRuntime.importType(appliedType)
+          runtimeTools.inferImplicitValue(runtimeType) match {
+            case ru.EmptyTree =>
+              ReprsOf.empty
```

Comment:
Would falling back to the old logic of `runtimeTools.inferImplicitValue(ru.appliedType(ru.typeOf[ReprsOf[_]].typeConstructor, importToRuntime.importType(typ)))` be helpful at all here?

Reply:
Or if `global` can't find it then there's no way that `ru` can, I guess.

Reply:
The runtime toolbox is still inferring the implicit; the only difference is that we use `global` to go from i.e. `DataFrame` to `ReprsOf[DataFrame]` and then import that type, rather than importing `DataFrame` and using the runtime universe to construct the `ReprsOf[DataFrame]` type.

I think it might be some race condition with classloading in the runtime universe. I don't think falling back would be helpful (it seems quite reliable to do it this way)